Binary files asus_oled/.svn/wc.db and asus-oled-sg/.svn/wc.db differ
diff -rupN asus_oled/Makefile asus-oled-sg/Makefile
--- asus_oled/Makefile	2014-04-14 22:24:12.858147208 +0200
+++ asus-oled-sg/Makefile	2014-04-15 17:12:09.192267448 +0200
@@ -1,19 +1,13 @@
-ifneq ($(KERNELRELEASE),)
-
-obj-m := asus_oled.o
-
-else
-
 obj-m := asus_oled.o
 
 KVER  := $(shell uname -r)
 
-KDIR := /lib/modules/$(KVER)/build
-MDIR := /lib/modules/$(KVER)/extra/
+KDIR := $(PREFIX)/lib/modules/$(KVER)/build
+MDIR := $(PREFIX)/lib/modules/$(KVER)/extra/
 PWD := $(shell pwd)
 
 default:
-	$(MAKE) -C $(KDIR) SUBDIRS=$(PWD) modules
+	$(MAKE) -C /lib/modules/$(KVER)/build SUBDIRS=$(PWD) modules
 	#
 	#
 	# To compile Qt3 utility for easy device control and image conversion,
@@ -24,9 +18,8 @@ default:
 install:
 	install -d $(MDIR)
 	install -m 644 -c $(obj-m:.o=.ko) $(MDIR)
-	depmod -a
+	# depmod -a
 
 clean:
 	rm -rf *~ *.mod.c *.o *.ko .tmp_versions Module.symvers .$(obj-m:.o=)* modules.order
 
-endif
diff -rupN asus_oled/asus_oled.c asus-oled-sg/asus_oled.c
--- asus_oled/asus_oled.c	2014-04-14 22:24:12.761481065 +0200
+++ asus-oled-sg/asus_oled.c	2013-12-15 04:42:21.000000000 +0100
@@ -731,26 +731,26 @@ static ssize_t version_show(struct class
 
 static CLASS_ATTR(version, S_IRUGO, version_show, NULL);
 
-static int __init asus_oled_init(void)
+static int __init asus_oled_init(struct asus_oled_dev *odev)
 {
 	int retval = 0;
 	oled_class = class_create(THIS_MODULE, ASUS_OLED_UNDERSCORE_NAME);
 
 	if (IS_ERR(oled_class)) {
-		err("Error creating " ASUS_OLED_UNDERSCORE_NAME " class");
+		dev_err(&odev->udev->dev, "Error creating " ASUS_OLED_UNDERSCORE_NAME " class\n");
 		return PTR_ERR(oled_class);
 	}
 
 	retval = class_create_file(oled_class, &class_attr_version);
 	if (retval) {
-		err("Error creating class version file");
+		dev_err(&odev->udev->dev, "Error creating class verion file\n");
 		goto error;
 	}
 
 	retval = usb_register(&oled_driver);
 
 	if (retval) {
-		err("usb_register failed. Error number %d", retval);
+		dev_err(&odev->udev->dev, "Error number %d\n", retval);
 		goto error;
 	}
 
